classes/Combination.php

```diff
@@ -123,7 +123,7 @@ public function delete()
         }
 
         // Removes the product from StockAvailable, for the current shop
-        StockAvailable::removeProductFromStockAvailable((int) $this->id_product, (int) $this->id);
+        StockAvailable::removeProductFromStockAvailable((int) $this->id_product, (int) $this->id, $this->id_shop);
```

Comment:
Im afraid these changes will break something in legacy pages :sweat:

Reply:
Instead of only using one shopId we should handle a list of shop Ids, this will be important for the All shops use case
```php
        if (count($this->id_shop_list)) {
            $id_shop_list = $this->id_shop_list;
        } else {
            $id_shop_list = Shop::getContextListShopID();
        }

        if (!empty($id_shop_list)) {
            foreach ($id_shop_list as $shopId) {
                StockAvailable::removeProductFromStockAvailable((int) $this->id_product, (int) $this->id, $shopId);
            }
        } else {
            StockAvailable::removeProductFromStockAvailable((int) $this->id_product, (int) $this->id);
        }
```